packages/jest-fake-timers/src/FakeTimersLolex.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  withGlobal as lolexWithGlobal,
+  LolexWithContext,
+  InstalledClock,
+} from 'lolex';
+import {formatStackTrace, StackTraceConfig} from 'jest-message-util';
+
+export default class FakeTimers {
+  private _clock!: InstalledClock;
+  private _config: StackTraceConfig;
+  private _fakingTime: boolean;
+  private _global: NodeJS.Global;
+  private _lolex: LolexWithContext;
+  private _maxLoops: number;
+
+  constructor({
+    global,
+    config,
+    maxLoops,
+  }: {
+    global: NodeJS.Global;
+    config: StackTraceConfig;
+    maxLoops?: number;
+  }) {
+    this._global = global;
+    this._config = config;
+    this._maxLoops = maxLoops || 100000;
+
+    this._fakingTime = false;
+    this._lolex = lolexWithGlobal(global);
```

Comment:
Wouldn't it work for you to just polyfill `queueMicrotask` in a setup file, then Lolex would pick it up and fake it?

---

We discussed this (passing it what to mock) briefly in the fall. IIRC the argument against allow you to pass in args was that you shouldn't be able to opt _out_ of certain methods - that would allow you to write tests in a way that would not reflect at all how it would run in the real world (beyond the fact you're faking time at all).

Reply:
I think a dummy `queueMicrotask` in an environment would suffice, yeah. It'd feel a bit weird because you don't actually need to even attempt to polyfill it - just having that property on the object is enough! But if I let little things like that bother me, coding in JavaScript would drive me insane...

Being hard to use wrong is a good goal. And there's already an escape hatch in manually setting up Lolex in whatever crazy way you want it, so this isn't critical or anything.